### PR TITLE
fix: NavigationFooter links pull paginated title from nav metadata

### LINKED ISSUE #451: "next" footer links don't pull title from source
Instead of pulling the title attribute from `next` target in nav metadata, the link generator seems to just strip the file extension and `-`. This looks weird if the title is different than the title generated from the link.

https://github.com/reactjs/reactjs.org/blob/184a7ef9476c524df0ff82362257c02bfbe8698d/src/templates/components/NavigationFooter/NavigationFooter.js#L89

EDIT: found an existing TODO for this 😄 

https://github.com/reactjs/reactjs.org/blob/184a7ef9476c524df0ff82362257c02bfbe8698d/src/components/MarkdownPage/MarkdownPage.js#L125-L132

### PR DESCRIPTION
**Pull request by raunofreiberg**
Fixes #451.

**Successfully ran:**
- yarn check-all

### CODE REVIEW

File: src/components/MarkdownPage/MarkdownPage.js [hunk 1]
```diff
@@ -31,6 +31,21 @@ type Props = {
   titlePostfix: string,
 };
 
+const getPaginatedTitle = (
```
Comment (alexkrolick) [inline src/components/MarkdownPage/MarkdownPage.js]: Looks great! My only nit is that this method name returns an  object, not just the title string, so a name like `getPageById` might be better than `getPaginatedTitle`.

Comment (raunofreiberg) [reply src/components/MarkdownPage/MarkdownPage.js]: I like this method name much better. Renamed in last commit. Thanks!

**Comment by raunofreiberg** [comment]
Found a small bug with the template file, more specifically I forgot to append the `.html` extension back to the remapped URL of the page. Fixed in last commit.

**Comment by raunofreiberg** [comment]
@alexkrolick anything else you'd like me to do with this PR? :-)

**Comment by gaearon** [comment]
Yay, thank you for fixing! This was super annoying.

**Comment by raunofreiberg** [comment]
My pleasure :-)

**Pull request merged**